pallets/grants/src/tests.rs

```diff
@@ -252,51 +281,124 @@ fn cancel_clears_storage() {
+		assert!(!<VestingSchedules<Runtime>>::contains_key(BOB::get()));
 	});
 }
 
 #[test]
 fn cancel_tolerates_corrupted_state() {
 	ExtBuilder::default().one_hundred_for_alice().build().execute_with(|| {
-		let schedule = VestingSchedule {
+		let allice_vesting_to_bob_schedule = VestingSchedule {
 			start: 0u64,
 			period: 10u64,
 			period_count: 2u32,
 			per_period: 10u64,
 		};
-		assert_ok!(Vesting::add_vesting_schedule(Origin::signed(ALICE), BOB, schedule));
+
+		// Initial Balance Status
+		assert_eq!(mock::balances(&ALICE::get()), (100, 0));
+		assert_eq!(PalletBalances::total_balance(&ALICE::get()), 100);
+
+		assert_eq!(mock::balances(&BOB::get()), (0, 0));
+		assert_eq!(PalletBalances::total_balance(&BOB::get()), 0);
+
+		assert_eq!(mock::balances(&CancelOrigin::get()), (0, 0));
+		assert_eq!(PalletBalances::total_balance(&CancelOrigin::get()), 0);
+
+		assert!(!<VestingSchedules<Runtime>>::contains_key(BOB::get()));
+
+		let ans = Vesting::add_vesting_schedule(
+			Origin::signed(ALICE::get()),
+			BOB::get(),
+			allice_vesting_to_bob_schedule.clone(),
+		);
+		assert_ok!(ans);
+		assert!(<VestingSchedules<Runtime>>::contains_key(BOB::get()));
+
+		// Event & Balance Status
+		let mut expected = vec![Event::VestingScheduleAdded(
+			ALICE::get(),
+			BOB::get(),
+			VestingSchedule {
+				start: 0,
+				period: 10,
+				period_count: 2,
+				per_period: 10,
+			},
+		)];
+		assert_eq!(context_events(), expected);
+
+		assert_eq!(mock::balances(&ALICE::get()), (80, 0));
+		assert_eq!(PalletBalances::total_balance(&ALICE::get()), 80);
+
+		assert_eq!(mock::balances(&BOB::get()), (20, 20));
+		assert_eq!(PalletBalances::total_balance(&BOB::get()), 20);
+
+		assert_eq!(mock::balances(&CancelOrigin::get()), (0, 0));
+		assert_eq!(PalletBalances::total_balance(&CancelOrigin::get()), 0);
 
 		// We also add some vesting schedules without any balances to simulate
 		// a corrupted / badly canceled state.
-		VestingSchedules::<Runtime>::mutate(BOB, |s| {
-			s.push(VestingSchedule {
-				start: 0u64,
-				period: 10u64,
-				period_count: 2u32,
-				per_period: 1_000u64, // definitely too much money
-			})
+		let bob_modified_vesting_schedule = VestingSchedule {
+			start: 0u64,
+			period: 10u64,
+			period_count: 2u32,
+			per_period: 1_000u64, // definitely too much money
+		};
+
+		let ans = <VestingSchedules<Runtime>>::try_mutate(BOB::get(), |s| -> Result<(), ()> {
+			s.try_push(bob_modified_vesting_schedule.clone())
 		});
+		assert_ok!(ans);
+
+		assert!(<VestingSchedules<Runtime>>::contains_key(BOB::get()));
+
+		if let Ok(schedule_from_chain) = <VestingSchedules<Runtime>>::try_get(BOB::get()) {
+			assert_eq!(schedule_from_chain.len(), 2);
+
+			assert_eq!(schedule_from_chain[0], allice_vesting_to_bob_schedule);
+			assert_eq!(schedule_from_chain[1], bob_modified_vesting_schedule);
+		} else {
+			#[cfg_attr(tarpaulin, ignore)]
+			panic!("Code expected to be dead is alive");
```

Comment:
Looks like the ignore config hasn't silenced the codecov complaint.

Reply:
Looks like line got removed.